src/DebugBar/Resources/debugbar.js

```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
 
     });
 
+    // ============= Handling Dragging Behavior ============
+    var storedPosX = localStorage.getItem('phpdebugbarPositionX') || 0;
+    var screenWidth = $(window).width();
+
+    $(document).on('click', '.phpdebugbar', function () {
+        let isClosed = $('.phpdebugbar').hasClass('phpdebugbar-closed');
+        if (!isClosed) {
+            // Save current position to localStorage when closing the debugbar
+            storedPosX = localStorage.getItem('phpdebugbarPositionX');
+            $('.phpdebugbar').css('left', '0px');
+        } else {
+            // Restore saved position from localStorage when opening the debugbar
+            localStorage.setItem('phpdebugbarPositionX', storedPosX);
+            $('.phpdebugbar').css('left', storedPosX + 'px');
+        }
+    });
+
+    $(document).ready(function () {
+        // Check local storage for saved position
+        var savedPosX = localStorage.getItem('phpdebugbarPositionX');
+        var phpdebugbarIsOpen = localStorage.getItem('phpdebugbar-open');
+
+        if (savedPosX !== null && phpdebugbarIsOpen == 0 && screenWidth > savedPosX) {
+            $('.phpdebugbar').css('left', savedPosX + 'px');
```

Comment:
Calling to search the entire document for classes like `.phpdebugbar` would be avoided if the class properties are reused, like `this.$el.css('left', savedPosX + 'px')`
https://github.com/maximebf/php-debugbar/blob/c9cb28d5ee6f657fcc48440f38db8bbd140bef83/src/DebugBar/Resources/debugbar.js#L488